src/Button.js

```diff
@@ -13,27 +12,25 @@ function fontSize({size = '14px', ...props}) {
   }
 }
 
-const ButtonBase = ({is: Tag, onClick, disabled, theme, ...rest}) => {
-  return <Tag disabled={disabled} onClick={disabled ? undefined : onClick} {...rest} />
-}
-
-const Button = styled(ButtonBase)`
+const Button = styled.button.attrs(props => ({
+  onClick: (props.disabled ? undefined : props.onClick),
+  className: (props.disabled ? 'disabled' : '')
```

Comment:
Will this interfere with any styled-components classNames or classNames passed by the user?

Reply:
Maybe it'd be better to do something similar to what you did [here](https://github.com/primer/components/pull/423#discussion_r265729558). Something like:
```js
className: classnames(props.className, props.disabled && 'disabled')
```

Reply:
Meh, we actually explicitly don't want people to add their own class names to our components. If custom CSS really needs to be added, users can use the `css` prop